DXRVanilla/DeusEx/Classes/Weapon.uc

```diff
@@ -653,6 +677,109 @@ function TravelPostAccept()
     }
 }
 
+simulated function bool IsWeaponTexture(int i)
+{
+    local Texture thisTex;
+
+    thisTex = GetMeshTexture(i);
+    if (thisTex==None){
+        return false;
+    } else if (InStr(string(thisTex),"MaskTex")!=-1){
+        return false;
+    } else if (InStr(string(thisTex),"WeaponHandsTex")!=-1){
+        return false;
+    } else if (InStr(string(thisTex),"SFX")!=-1){
+        return false;
+    } else if (InStr(string(thisTex),"MapTex")!=-1){
+        return false;
+    } else {
+        return true;
+    }
+}
+
+simulated event RenderOverlays( canvas Canvas )
+{
+    local Texture origTex[8];
+    local int texLoc;
+    Super.RenderOverlays(Canvas);
+
+    //Draw an indication that the weapon still has a shot in progress
+    //if (bFiring && !IsAnimating() && PlayerPawn(Owner)!=None){
+    //bPointing seems to be updated basically the same as bFiring, except it works for melee as well
+    if (GetStateName()=='NormalFire' && !IsAnimating() && !bAutomatic && PlayerPawn(Owner)!=None){
```

Comment:
I wonder if we should remove the `!IsAnimating()` check so it's red during the animation too?